src/classes/aaircraftentry.h

```diff
@@ -1,3 +1,20 @@
+/*
+ *openPilot Log - A FOSS Pilot Logbook Application
+ *Copyright (C) 2020  Felix Turowsky
+ *
+ *This program is free software: you can redistribute it and/or modify
+ *it under the terms of the GNU General Public License as published by
+ *the Free Software Foundation, either version 3 of the License, or
+ *(at your option) any later version.
+ *
+ *This program is distributed in the hope that it will be useful,
+ *but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *GNU General Public License for more details.
+ *
+ *You should have received a copy of the GNU General Public License
+ *along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
```

Comment:
Putting the license on every file is just too much.

Reply:
It's a [requirement](https://www.gnu.org/licenses/gpl-faq.html#NoticeInSourceFile) for the GPL license to be included in every source file. It is not the whole license but the [preamble](https://www.gnu.org/licenses/gpl-faq.html#GPLOmitPreamble) which must not be omitted.

Reply:
Nothing in both links says that you have to put the license/preaamble in every file.

The first link says that every file should have a *notice* of the license its under. Havent seen it specified, could be just one line stating that the current file is under GPL license for the case someone separates it from the source.

The second link says that the *license* should not omit *any* parts of the gpl.

Reply:
> Havent seen it specified,

If you read on a bit more on the page, you will find explanations going more into details on how those terms are to be interpreted.

You can find an explanation on the format of the notice [here](https://www.gnu.org/licenses/gpl-howto.html) under the header license notices.

That this license notice (and the copyright notice) can not be omitted is stated in the first link.

Reply:
This is ludicrous... why pollute every file with 40 lines. I get the "separated from source" part but cmon

Reply:
It's not ideal, but I don't make these rules, we'll just have to comply with it.

I chose GPL because Qt is a commercial framework. It can be used under open source licensing, but there are certain [restrictions](https://www.qt.io/licensing/) and the project using Qt under an open source license needs to be either GPL or LGPL licensed. As for what has to be included in the license notice, there is no difference between the two, but the GPL is more expansive and detailed, which leads to using Qt under the GPL having [less restrictions](https://www.qt.io/licensing/open-source-lgpl-obligations#lgpl).